src/main/java/org/opensearch/ad/rest/RestAnomalyDetectorJobAction.java

```diff
@@ -16,60 +16,138 @@
     }
 
-    @Override
     public String getName() {
         return AD_JOB_ACTION;
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+    public List<RouteHandler> routeHandlers() {
+        return ImmutableList
+            .of(
+                // start AD Job
+                new RouteHandler(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, DETECTOR_ID, START_JOB),
+                    handleRequest
+                )
+            );
+    }
+
+    private Function<RestRequest, ExtensionRestResponse> handleRequest = (request) -> {
+        try {
+            return prepareRequest(request);
+        } catch (Exception e) {
+            return exceptionalRequest(request, e);
+        }
+    };
+
+    private void registerJobDetails() throws IOException {
+
+        XContentBuilder requestBody = JsonXContent.contentBuilder();
+        requestBody.startObject();
+        requestBody.field(GetJobDetailsRequest.JOB_INDEX, AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX);
+        requestBody.field(GetJobDetailsRequest.JOB_TYPE, AnomalyDetectorExtension.AD_JOB_TYPE);
+        requestBody.field(GetJobDetailsRequest.JOB_PARAMETER_ACTION, ADJobParameterAction.class.getName());
+        requestBody.field(GetJobDetailsRequest.JOB_RUNNER_ACTION, ADJobRunnerAction.class.getName());
+        requestBody.field(GetJobDetailsRequest.EXTENSION_UNIQUE_ID, extensionsRunner.getUniqueId());
+        requestBody.endObject();
+
+        Request request = new Request("PUT", String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_job_details"));
+        request.setJsonEntity(Strings.toString(requestBody));
+
+        Response response = client.performRequest(request);
+        this.jobDetailsAreRegistered = RestStatus.fromCode(response.getStatusLine().getStatusCode()) == RestStatus.OK ? true : false;
+        LOG.info("Job Details Registered : " + jobDetailsAreRegistered);
+    }
+
+    protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOException {
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
 
+        // Ensure job details are registered with Job Scheduler prior to creating a job
+        if (!jobDetailsAreRegistered) {
+            registerJobDetails();
+        }
+
         String detectorId = request.param(DETECTOR_ID);
         long seqNo = request.paramAsLong(IF_SEQ_NO, SequenceNumbers.UNASSIGNED_SEQ_NO);
         long primaryTerm = request.paramAsLong(IF_PRIMARY_TERM, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
-        boolean historical = request.paramAsBoolean("historical", false);
-        String rawPath = request.rawPath();
+        // Passed false until historical analysis workflow is enabled
+        boolean historical = false;
```

Comment:
This should take the default value as `false`
```suggestion
        boolean historical =  request.paramAsBoolean("historical", false);
```